ykpack/src/types.rs

```diff
@@ -102,19 +114,112 @@ impl Display for Statement {
     }
 }
 
+/// A place for storing things.
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum Place {
+    Base(PlaceBase),
+    Projection(Projection<Box<Place>, LocalIndex>),
+}
+
+/// The "base" of a place projection.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum PlaceBase {
     Local(LocalIndex),
-    Unimplemented, // FIXME
+    Static(DefId),
+    Promoted(PromotedIndex),
 }
 
+/// A projection (deref, index, field access, ...).
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
-pub enum Rvalue {
+pub struct Projection<B, V> {
+    pub base: B,
+    pub elem: ProjectionElem<V>,
+}
+
+/// Describes a projection operation upon a projection base.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum ProjectionElem<V> {
+    Unimplemented(PhantomData<V>), // FIXME
```

Comment:
Why fixme?

Reply:
There's supposed to be another structure in there, but I don't want to implement it just yet.

The missing struct is parameterised by V.